### PR TITLE
AC-921: Billling Page should allow manually billed customers to switch to self-serve

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed
 - Manually billed banner brings to credit card addition form
 - Form transitions customer into self serve billing

### How To Test
 - Create a new account (w/o selecting plan)
 - Transition to a manually billed customer (check the admin docs for billing API)
 - Go to /account/billing
   -Verify a manually billed banner exists
- Check that the link navigates to /account/billing/enable-automatic
 - Check that submission works and navigates back to /account/billing
 - Should now show billing summary rather than manually billed banner

### To Do
- [x] Wait for CORS fix in API
- [x] Wait for fix in delay of flag
- [x] Update check for manually billed to be in billing/subscription
- [ ] Address feedback

### Notes
Waiting for update in API for flag change

### CODE REVIEW

File: src/pages/billing/components/ManuallyBilledBanner.js [hunk 1]
```diff
@@ -41,17 +44,35 @@ const ManuallyBilledBanner = ({
...
+          }.
+        </p>
+        {onZuoraPlan && (
+          <p>
+            Enable automatic billing to self-manage your plan and add-ons.
+          </p>
+        )}
+      </Banner>
+      <Modal>
+        <UpdatePaymentForm />
```
Comment (nicklemmon) [inline src/pages/billing/components/ManuallyBilledBanner.js]: This payment update takes quite awhile, so making sure the button transitions to 'Loading...' or something while that's occurring would be a big UX improvement. Right now everything just disables.

File: src/actions/billing.js [hunk 2]
```diff
@@ -1,46 +1,124 @@
...
+  //TODO: Replace with sparkpostApiRequest
+  return mockThunk(
+    {
+      type: 'GET_BUNDLES',
+      meta: {
+        method: 'GET',
+        url: '/v1/billing/bundles',
+      },
+    },
+    { data: MOCK_BUNDLES }, //TODO: Delete mock response
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/actions/billing.js]: Actually, looks like it's being used by `NewChangePlanForm` - I assume that's behind a feature flag? If so, just ignore my previous comment.

File: src/selectors/tests/accountBillingInfo.test.js [hunk 3]
```diff
@@ -190,7 +189,7 @@ describe('plan selector', () => {
     });
 
     it('should return active paid plans', () => {
-      state.account.subscription.self_serve = false;
+      state.billing.subscription.type = 'manual';
       expect(billingInfo.selectAvailablePlans(state)).toMatchSnapshot();
```
Comment (nicklemmon) [inline src/selectors/tests/accountBillingInfo.test.js]: Any interest in updating these to explicit tests? I'm just not sure what I'm supposed to be looking for with the snapshots.

File: src/helpers/conditions/account.js [hunk 4]
```diff
@@ -10,8 +10,9 @@ export const hasStatus = status => ({ account }) => account.status === status;
 export const hasStatusReasonCategory = category => ({ account }) =>
   account.status_reason_category === category;
 export const isSuspendedForBilling = all(hasStatus('suspended'), hasStatusReasonCategory('100.01'));
+const SELF_SERVE = ['active', 'inactive', 'none'];
```
Comment (nicklemmon) [inline src/helpers/conditions/account.js]: A more clear name might be `SELF_SERVE_ACCOUNT_TYPES`.

File: src/helpers/conditions/tests/account.test.js [hunk 5]
```diff
@@ -191,38 +169,58 @@ describe('Condition: hasUiOption', () => {
...
+        options: { option: false },
+        defaultVal: true,
+        result: false,
+      },
+      'Account option precedence: true/true=true': {
+        options: { option: true },
+        defaultVal: true,
+        result: true,
+      },
+      // Default used iff option is missing
```
Comment (nicklemmon) [inline src/helpers/conditions/tests/account.test.js]: Typo here (even though I know this is a Prettier re-formatting situation, may as well fix it)

File: src/helpers/conditions/account.js [hunk 6]
```diff
@@ -1,27 +1,27 @@
...
-export const subscriptionSelfServeIsTrue = ({ account }) => _.get(account, 'subscription.self_serve', false);
+export const onPlanWithStatus = status => ({ accountPlan }) => accountPlan.status === status;
+export const onServiceLevel = level => ({ account }) => account.service_level === level;
+export const isEnterprise = any(onPlan('ent1'), onServiceLevel('enterprise'));
+export const hasStatus = status => ({ account }) => account.status === status;
+export const hasStatusReasonCategory = category => ({ account }) =>
+  account.status_reason_category === category;
+export const isSuspendedForBilling = all(hasStatus('suspended'), hasStatusReasonCategory('100.01'));
+const SELF_SERVE = ['active', 'inactive', 'none'];
+export const subscriptionSelfServeIsTrue = ({ subscription }) =>
```
[the remark this answers is not in the document]
Comment (PatrickSison) [reply src/helpers/conditions/account.js]: agreed 👍

File: src/pages/billing/components/ManuallyBilledBanner.js [hunk 7]
```diff
@@ -41,17 +44,35 @@ const ManuallyBilledBanner = ({
...
+      </Modal>
```
[the remark this answers is not in the document]
Comment (PatrickSison) [reply src/pages/billing/components/ManuallyBilledBanner.js]: I think I added this in at first, then found the other form and forgot to remove this. Removing now.

**Pull request merged**